src/sentry/runner/initializer.py

```diff
@@ -215,10 +221,8 @@ def apply_legacy_settings(settings):
 
     # Set ALLOWED_HOSTS if it's not already available
     if not settings.ALLOWED_HOSTS:
-        from urlparse import urlparse
-        urlbits = urlparse(settings.SENTRY_URL_PREFIX)
-        if urlbits.hostname:
-            settings.ALLOWED_HOSTS = (urlbits.hostname,)
+        from .hacks import AllowedHosts
+        settings.ALLOWED_HOSTS = AllowedHosts()
```

Comment:
Also, just to answer:

>  (or listen to changes on system.url-prefix and define ALLOWED_HOSTS from it

This won't work since we have many processes to coordinate. This would be great to wrap in a Django signal or something and on `option.set()`, update. :) But we'd have to do something more complex. Like, on a `get()`, if the value has changed, trigger the signal. Not just on `set()`.